kura/org.eclipse.kura.rest.network.configuration.provider/src/main/java/org/eclipse/kura/internal/rest/network/configuration/NetworkConfigurationRestService.java

```diff
@@ -273,4 +276,32 @@ private boolean isNetworkConfigurationPid(String pid) {
         }
         return result;
     }
+
+    private void updateNetInterfaces(Map<String, Object> configurationProperties, String pid) throws KuraException {
+        if (pid.equals(NETWORK_CONFIGURATION_SERVICE_PID)) {
+            Set<String> interfaceNames = new HashSet<>();
+            configurationProperties.keySet().forEach(key -> {
+                Optional<String> interfaceName = parseInterfaceName(key);
+                interfaceName.ifPresent(interfaceNames::add);
+            });
+            Map<String, Object> properties = this.configurationService.getComponentConfiguration(pid)
+                    .getConfigurationProperties();
+            String netInterfaces = (String) properties.get("net.interfaces");
+            List<String> netInterfacesList = Arrays.asList(netInterfaces.split(","));
+            for (String name : interfaceNames) {
+                if (!netInterfacesList.contains(name)) {
+                    netInterfaces += "," + name;
+                }
+            }
+            configurationProperties.put("net.interfaces", netInterfaces);
+        }
+    }
+
+    private Optional<String> parseInterfaceName(String key) {
+        Optional<String> interfaceName = Optional.empty();
+        if (key.startsWith("net.interface.") && key.contains(".config.")) {
+            interfaceName = Optional.of(key.substring(14, key.indexOf(".config.")));
+        }
```

Comment:
Since we already use Apache Common's StringUtils in our code base, wouldn't it be better [to use `StringUtils.substringBetween`](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#substringBetween(java.lang.String,java.lang.String,java.lang.String)) for readability?